packages/croncat-sdk-agents/src/msg.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Timestamp, Uint128, Uint64};
+
+use crate::types::AgentStatus;
+
+#[cw_serde]
+pub struct AgentContractInstantiateMsg {}
+
+#[cw_serde]
+pub enum AgentExecuteMsg {
+    RegisterAgent {
+        payable_account_id: Option<String>,
+        cost: u128,
+    },
+    UpdateAgent {
+        payable_account_id: String,
+    },
+    CheckInAgent {},
+    UnregisterAgent {
+        from_behind: Option<bool>,
+    },
+    WithdrawReward {},
+}
+
+#[cw_serde]
+pub enum AgentQueryMsg {
```

Comment:
```suggestion
#[cw_serde]
#[derive(QueryResponses)]
pub enum AgentQueryMsg {
```
Let's use `QueryResponse` for easier schema handle!